### PR TITLE
Targeting

### PR DESCRIPTION
**Pull request by Rik-de-Kort**
Accidentally did this work on `main`, but here it is as a PR. :)

I think this could definitely use some review, but at least we have a working implementation.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -78,12 +81,29 @@ impl TestCase {
     /// Return an integer in the range [0, n]
     fn choice(&mut self, n: u64) -> Result<u64, Error> {
         if self.choices.len() < self.prefix.len() {
-            self.forced_choice(self.prefix[self.choices.len()])
+            let preordained = self.prefix[self.choices.len()];
+            if preordained > n {
+                Err(Error::Invalid)
+            } else {
+                self.forced_choice(self.prefix[self.choices.len()])
```
Comment (Stranger6667) [inline src/main.rs]: Would this work?

```suggestion
                self.forced_choice(preordained)
```

File: src/main.rs [hunk 2]
```diff
@@ -353,35 +373,149 @@ impl TestState {
...
+                        }
+                        k /= 2;
+                    }
                 }
             }
-            Err(_) => false,
         }
     }
 
+    /// Main test runner: insrtantiate a TestState and then use this to run.
```
Comment (Stranger6667) [inline src/main.rs]:
```suggestion
    /// Main test runner: instantiate a TestState and then use this to run.
```

**Review by Stranger6667 — CHANGES_REQUESTED** [review]
Looks good to me, only two minor things :)

File: src/main.rs (hunk 1 above)
Comment (Rik-de-Kort) [reply src/main.rs]: That would most certainly work

**Review by Stranger6667 — APPROVED** [review]

**Pull request merged**